src/libs/Navigation/AppNavigator/ReportScreenIDSetter.ts

```diff
@@ -1,91 +1,66 @@
 import {useEffect} from 'react';
-import PropTypes from 'prop-types';
-import lodashGet from 'lodash/get';
-import {withOnyx} from 'react-native-onyx';
+import {OnyxCollection, OnyxEntry, withOnyx} from 'react-native-onyx';
 import ONYXKEYS from '../../../ONYXKEYS';
 import * as ReportUtils from '../../ReportUtils';
-import reportPropTypes from '../../../pages/reportPropTypes';
-import {withNavigationPropTypes} from '../../../components/withNavigation';
 import * as App from '../../actions/App';
 import usePermissions from '../../../hooks/usePermissions';
 import CONST from '../../../CONST';
 import Navigation from '../Navigation';
+import {Policy, Report} from '../../../types/onyx';
+import {ReportScreenWrapperProps} from './types';
 
-const propTypes = {
+type ReportScreenIDSetterComponentProps = {
     /** Available reports that would be displayed in this navigator */
-    reports: PropTypes.objectOf(reportPropTypes),
+    reports: OnyxCollection<Report>;
 
     /** The policies which the user has access to */
-    policies: PropTypes.objectOf(
-        PropTypes.shape({
-            /** The policy name */
-            name: PropTypes.string,
+    policies: OnyxCollection<Policy>;
 
-            /** The type of the policy */
-            type: PropTypes.string,
-        }),
-    ),
-
-    isFirstTimeNewExpensifyUser: PropTypes.bool,
-
-    /** Navigation route context info provided by react navigation */
-    route: PropTypes.shape({
-        /** Route specific parameters used on this screen */
-        params: PropTypes.shape({
-            /** If the admin room should be opened */
-            openOnAdminRoom: PropTypes.bool,
-
-            /** The ID of the report this screen should display */
-            reportID: PropTypes.string,
-        }),
-    }).isRequired,
-
-    ...withNavigationPropTypes,
+    isFirstTimeNewExpensifyUser: OnyxEntry<boolean>;
 };
 
-const defaultProps = {
-    reports: {},
-    policies: {},
-    isFirstTimeNewExpensifyUser: false,
-};
+type ReportScreenIDSetterProps = ReportScreenIDSetterComponentProps & ReportScreenWrapperProps;
 
 /**
  * Get the most recently accessed report for the user
- *
- * @param {Object} reports
- * @param {Boolean} ignoreDefaultRooms
- * @param {Object} policies
- * @param {Boolean} isFirstTimeNewExpensifyUser
- * @param {Boolean} openOnAdminRoom
- * @returns {Number}
  */
-const getLastAccessedReportID = (reports, ignoreDefaultRooms, policies, isFirstTimeNewExpensifyUser, openOnAdminRoom) => {
+const getLastAccessedReportID = (
+    reports: OnyxCollection<Report> | Report[],
```

Comment:
Are you sure that it can be also `Report[]`?

Reply:
@kubabutkiewicz It's then passed to the ReportUtils function which has array in types https://github.com/Expensify/App/blob/main/src/libs/ReportUtils.js#L534
But I'll double-check and maybe remove the array here, especially after ReportUtils migration

Reply:
I am currently migratiing this and I changed it to just `OnyxCollection<Report>` because what I saw it was only recieving that but also will tripple check :D